### ISSUE TITLE
testing of ani fails for checking AngularSymmetryFunction on Mac OS.

### ISSUE THREAD

**Issue by chrisiacovella** [choderalab/modelforge#193]
Currently CI is failing for `test_ani.py::test_compare_angular_symmetry_features` on MacOS.  These tests still pass on linux. This tests the AngularSymmetryFunction in `modelforge.potential.utils`.  This failure only started after it was switched from calling the original implementation in torchani directly to using precomputed values. 

If I were to recompute the angular feature vector using torchani on macOS and compare to the output of modelforge, I get matching results; these values differ from linux.  Flattening the results and summing leads to matching results; on closer inspection, the values appear the same just in a different order.  

The issue appears related to the use of `sort` in `triple_by_molecule` in potential/utils.py.  Specifically, this line:
https://github.com/choderalab/modelforge/blob/56093eae69a36427c99272ef61db136abd0fffde/modelforge/potential/utils.py#L105C1-L106C1

```
    sorted_ai1, rev_indices = ai1.sort()
```

While the contents of the `rev_indices` tensor are the same on both platforms, the order is different. Modifying this to use the stable mode fixing this issue.
```
    sorted_ai1, rev_indices = ai1.sort(stable=True)
```

Interestingly, even with stable=False by default, MacOS version appears to produce the same output as stable=True, at least for the test case.   

Since ultimately the order of the indices should not impact the calculation here, we could leave the default in place (stable=False) and simply consider the sum of the angular_feature_vectors in the test.  This is likely favorable as setting stable=True will be slower (although it is unclear to me if substantially slower; I assume that depends upon the system size).

**Comment by wiederm** [comment]
Thanks for raising this! That is really interesting!

**Comment by chrisiacovella** [comment]
I decided just to compare the sum of the angular feature vector (for the reference and modelforge versions), as that should not depend on order (within reasonable tolerances) and should still be sensitive to catch if there is something wrong.    I've added in some notes to the code  both in the `triple_by_molecule` function and also in the `test_compare_angular_symmetry_features` function, so this isn't forgotten for the future. This is being fixed in #189
